File: 03-cpp11/base64url.cpp
/*
####################################
#
# --- BASE64URL.C++ ---
#
####################################

 Example usage:

    gcc base64url.cpp; ./a.out > myfile0001.txt

 Then to use Qrencode or Zint:

    qrencode -8 -o qr-myfile-0001.png -r myfile0001.txt
    zint -b 58 --eci=26 -o qr-myfile-0001.png -i myfile0001.txt

*/

#include <cstdlib>

int main(int argc, char *argv[])
{
// Change the data to suit your needs:
  std::system("printf \"20231023-192045@kevinleake01\" | basenc --base64url");
}

File: 04-golang/base64url.go
/*
----- BASE64URL.GO -----

To run this program:

  go run base64url.go > myfile-0001.txt

Then to use Qrencode:

  qrencode -8 -o qr-myfile-0001.png -r myfile-0001.txt

Or Zint:

  zint -b 58 --eci=26 -o qr-myfile-0001.png -i myfile-0001.txt

*/
package main

import (
  b64 "encoding/base64"
    "fmt"
)

func main() {

// Change the data to suit your needs:
  data := "20240503-175240@kevinleake01"

  uEnc := b64.URLEncoding.EncodeToString([]byte(data))
  fmt.Println(uEnc)
}